### ISSUE TITLE
Memory leak of detached DOM elements

### ISSUE THREAD

**Issue by jinh0** [DDMAL/Neon#857]
Related to #767 . I believe this is the main cause of Neon's high memory consumption.

## Explanation
**TL;DR:** It is not an issue with our cache. It's an issue with how JavaScript garbage collection works. We have a memory leak of detached DOM elements.

![Screen Shot 2022-05-30 at 3 49 58 PM](https://user-images.githubusercontent.com/40512164/171053692-1f64fbd0-9dfd-41f0-9078-ced597bf72e7.png)

JavaScript makes sure that you don't consume too much memory by cleaning up after you. Once you label some memory as garbage, as memory that's not needed, JavaScript collects that garbage and releases it from memory. As long as you are referencing some memory, it is not garbage.

Every time the user makes an edit, Neon rerenders the entire SVG, and we would expect the old rendered SVG to be thrown away as garbage. But we're not properly handling our old SVG, so JavaScript thinks it's not garbage. So these old SVGs are just kept in memory. After ~20 edits, there's ~250 MB of garbage memory being kept. The image I've attached is Chrome DevTools listing all the elements that should be thrown away but aren't.

Just to clarify, the SVG is different from the MEI files. The MEIs being stored on the undoStacks and redoStacks are fine. MEI is not something that gets rendered onto HTML. SVG is what is rendered on the HTML, and hence that's the issue.

What we need to do is make sure that we dereference these old elements properly, so that they are collected by our garbage collector. 

[More information on JavaScript memory leaks](https://developer.chrome.com/docs/devtools/memory-problems/)

**Comment by jinh0** [comment]
Closed as this will be fixed in #767 .
